be/src/vec/aggregate_functions/aggregate_function_linear_histogram.h

```diff
@@ -0,0 +1,243 @@
+struct AggregateFunctionLinearHistogramData {
+    // bucket key limits
+    const static int32_t MIN_BUCKET_KEY = std::numeric_limits<int32_t>::min();
+    const static int32_t MAX_BUCKET_KEY = std::numeric_limits<int32_t>::max();
+
+private:
+    // influxdb use double
+    double interval = 0;
+    double offset;
+    double lower; // not used yet
+    double upper; // not used yet
+    std::map<int32_t, size_t> buckets;
+
+public:
+    // reset
+    void reset() {
+        offset = 0;
+        interval = 0;
+        buckets.clear();
+    }
+
+    void set_parameters(double input_interval, double input_offset) {
+        interval = input_interval;
+        offset = input_offset;
+    }
+
+    // add
+    void add(const T& value, UInt32 scale) {
+        double val = 0;
+        if constexpr (IsDecimalNumber<T>) {
+            using NativeType = typename T::NativeType;
+            val = static_cast<double>(value.value) / decimal_scale_multiplier<NativeType>(scale);
+        } else {
+            val = static_cast<double>(value);
+        }
+        double key = std::floor((val - offset) / interval);
+        if (key <= MIN_BUCKET_KEY || key >= MAX_BUCKET_KEY) {
+            throw doris::Exception(ErrorCode::INVALID_ARGUMENT, "{} exceeds the bucket range limit",
+                                   value);
+        }
+        buckets[static_cast<int32_t>(key)]++;
+    }
+
+    // merge
+    void merge(const AggregateFunctionLinearHistogramData& rhs) {
+        if (rhs.interval == 0) {
+            return;
+        }
+
+        interval = rhs.interval;
+        offset = rhs.offset;
+
+        for (const auto& [key, count] : rhs.buckets) {
+            buckets[key] += count;
+        }
+    }
+
+    // write
+    void write(BufferWritable& buf) const {
+        write_binary(offset, buf);
+        write_binary(interval, buf);
+        write_binary(lower, buf);
+        write_binary(upper, buf);
+        write_binary(buckets.size(), buf);
+        for (const auto& [key, count] : buckets) {
+            write_binary(key, buf);
+            write_binary(count, buf);
+        }
+    }
+
+    // read
+    void read(BufferReadable& buf) {
+        read_binary(offset, buf);
+        read_binary(interval, buf);
+        read_binary(lower, buf);
+        read_binary(upper, buf);
+        size_t size;
+        read_binary(size, buf);
+        for (size_t i = 0; i < size; i++) {
+            int32_t key;
+            size_t count;
+            read_binary(key, buf);
+            read_binary(count, buf);
+            buckets[key] = count;
+        }
+    }
+
+    // insert_result_into
+    void insert_result_into(IColumn& to) const {
+        rapidjson::Document doc;
+        doc.SetObject();
+        rapidjson::Document::AllocatorType& allocator = doc.GetAllocator();
+
+        unsigned num_buckets =
+                buckets.empty() ? 0 : buckets.rbegin()->first - buckets.begin()->first + 1;
+        doc.AddMember("num_buckets", num_buckets, allocator);
```

Comment:
define and use const for all strings

Reply:
I have tried, but `rapidjson` api does not support const strings.